src/HotChocolate/AspNetCore/src/AspNetCore/Serialization/DefaultHttpRequestParser.cs

```diff
@@ -47,6 +51,88 @@ internal class DefaultHttpRequestParser : IHttpRequestParser
             CancellationToken cancellationToken) =>
             ReadAsync(stream, false, cancellationToken);
 
+        public async ValueTask<IReadOnlyList<GraphQLRequest>> ReadMultipartRequestAsync(
+            HttpRequest httpRequest,
+            CancellationToken cancellationToken)
+        {
+            // todo: The IFormCollection is convenient, but it requires us to work with strings instead of a stream
+            var formFeature = new FormFeature(httpRequest);
+            var form = await formFeature.ReadFormAsync(cancellationToken);
+
+            if (form.Count < 2)
+                throw new Exception("At least a 'operations' and 'map' field need to be present.");
+
+            if (!form.TryGetValue("operations", out StringValues operationsCollection) || !operationsCollection.TryPeek(out var operationsString))
+                throw new Exception("No 'operations' specified.");
```

Comment:
I have started in my local branch to build in proper request errors.

Reply:
We could copy the errors used in the [reference implementation](https://github.com/jaydenseric/graphql-upload/blob/master/public/processRequest.js).

Reply:
That would be a good idea.